src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs

```diff
@@ -347,14 +347,15 @@ private void OnBatchScopeDisposed()
                 }
 
                 var documentFileNamesAdded = ImmutableArray.CreateBuilder<string>();
-                var documentsToOpen = new List<(DocumentId, SourceTextContainer)>();
+                var documentsToOpen = new List<(DocumentId documentId, SourceTextContainer textContainer, bool isAdditionalDocument)>();
 
                 _workspace.ApplyBatchChangeToProject(Id, solution =>
                 {
                     solution = _sourceFiles.UpdateSolutionForBatch(
                         solution,
                         documentFileNamesAdded,
                         documentsToOpen,
```

Comment:
It seems we should just have a separate "documentsToOpen" and "additionalDocumentsToOpen" collection.